### PR TITLE
test: add sad paths for sign in with password identifier flow

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

- Add sad paths for sign in with password identifier integration tests
- refactor: align sie config reset logic for interaction api tests
### Todo
Add response status guard for all interaction related APIs, We need do that after all tests is ready since a single flow will not cover all interaction API calls.

## Testing

Integration tests

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [x] integration tests
- [ ] docs

OR

- [x] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/integration-tests/src/tests/api/interaction/sign-in-with-password-identifier/sad-path.test.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+  setEmailConnector,
+} from '#src/helpers/connector.js';
+import { expectRejects } from '#src/helpers/index.js';
+import { enableAllPasswordSignInMethods } from '#src/helpers/sign-in-experience.js';
+import { generateNewUser } from '#src/helpers/user.js';
+import { generateName, generatePassword } from '#src/utils.js';
+
+describe('Sign-in flow sad path using password identifiers', () => {
+  beforeAll(async () => {
+    await enableAllPasswordSignInMethods();
```
Comment (simeng-li) [inline packages/integration-tests/src/tests/api/interaction/sign-in-with-password-identifier/sad-path.test.ts:20]: should set the SIE exclusively under each test case? Feel like `enableAllPasswordSignInMethods` for this test suite is useless.

File: packages/integration-tests/src/tests/api/interaction/sign-in-with-password-identifier/sad-path.test.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    await setEmailConnector();
+  });
+
+  afterAll(async () => {
+    await clearConnectorsByTypes([ConnectorType.Sms, ConnectorType.Email]);
+  });
+
+  afterEach(async () => {
+    await enableAllPasswordSignInMethods();
+  });
```
Comment (simeng-li) [inline packages/integration-tests/src/tests/api/interaction/sign-in-with-password-identifier/sad-path.test.ts:32]: ditto

Comment (xiaoyijun) [reply packages/integration-tests/src/tests/api/interaction/sign-in-with-password-identifier/sad-path.test.ts]: updated.

File: packages/integration-tests/src/tests/api/interaction/sign-in-with-password-identifier/sad-path.test.ts (hunk 1 above)
Comment (xiaoyijun) [reply packages/integration-tests/src/tests/api/interaction/sign-in-with-password-identifier/sad-path.test.ts:14]: This flow test rely on `enableAllPasswordSignInMethods` sie settings.
Some special tests will setup their own configs and reset accordingly.

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**